Automattic/wp-calypso
# Domains: Update DomainItem component

#### Changes proposed in this Pull Request

This PR redesigns the domain item component used in the site domains management page ("Upgrades > Domains") and in the all domains management page. The old `DomainItem` component will be replaced by a new `DomainRow` component, which has a more tabular layout.

Current design:

<img width="1076" alt="Screen Shot 2021-10-25 at 22 48 39" src="https://user-images.githubusercontent.com/5324818/138794985-429ab2c1-f829-482b-9e38-eef946f9294b.png">

<img src="https://user-images.githubusercontent.com/5324818/138795183-b84f0912-3c22-4cf0-9c22-e4200da72b42.png">

New design:

<img width="1082" alt="Screen Shot 2021-10-25 at 22 37 30" src="https://user-images.githubusercontent.com/5324818/138794778-e432da1f-e534-4a20-bef3-ae7dc5f8ae6c.png">

<img src="https://user-images.githubusercontent.com/5324818/138795231-6f144251-f0fd-421a-92af-9a0df4dc8aa4.png">

This is part of the domain managements page redesign project, detailed in pcYYhz-m2-p2.

#### Testing instructions

This is best tested with `D68679-code` applied to your sandbox so that you can check domains in many different states.

Access the live Calypso link and go to "Upgrades > Domains". Ensure the page looks like the designs in the Figma project (`hlVh2q24ad6MCwlwNnE9PQ-fi`) and screenshots above. Please check the page in mobile view too.

If you had applied `D68679-code` to your sandbox, remove it in order to test the domain actions. Make sure all domain functionalities are working as they should - access domain details, change primary site address, click the add mailbox link, etc. Also verify if the informations being shown are correct: domain expiration or renewal date, auto-renew state, mailboxes, etc.
